### ISSUE TITLE
tombstone in entryrdn index with lmdb but not with bdb

### ISSUE THREAD

**Issue by progier389** [389ds/389-ds-base#5490]
**Issue Description**

There is a inconsistency 

**Package Version and Platform:**
 - Platform: Fedora
 - Package and version: main branch

**Steps to Reproduce**
1. Run following script
4. See error: nsuniqueid are present in mdb (before and after reindex but not in mdb)

**Expected results**
bdb and mdb should have the same result (no tombstone in entryrdn)

```
DBBASEDIR=/tmp/DB

doit()
{
    export NSSLAPD_DB_LIB=${1:?}
    mkm2  # create supplier1/supplier2 instances and setup replication

    ldapmodify -Y EXTERNAL -H ldapi://$S1 <<!
dn: ou=new_ou,ou=people,dc=example,dc=com
changetype: add
ou: new_ou
objectClass: organizationalunit

dn: uid=new_demo_user,ou=new_ou,ou=people,dc=example,dc=com
changetype: add
nsAccountLock: true
loginShell: /bin/false
homeDirectory: /var/empty
gidNumber: 99997
uidNumber: 99997
legalName: Demo User Name
displayName: Demo User
cn: New Demo User
uid: new_demo_user
objectClass: top
objectClass: nsPerson
objectClass: nsAccount
objectClass: nsOrgPerson
objectClass: posixAccount

dn: uid=demo_user,ou=people,dc=example,dc=com
changetype: modify
replace: description
description: $(date)

dn: uid=new_demo_user,ou=new_ou,ou=people,dc=example,dc=com
changetype: delete

dn: ou=new_ou,ou=people,dc=example,dc=com
changetype: delete
!

    cp -r $PREFIX/var/lib/dirsrv/slapd-supplier1/db $DBBASEDIR/db.$NSSLAPD_DB_LIB.br
    dsconf supplier1 backend index reindex userroot
    cp -r $PREFIX/var/lib/dirsrv/slapd-supplier1/db $DBBASEDIR/db.$NSSLAPD_DB_LIB.ar
    dbscan -D ${NSSLAPD_DB_LIB:?} -f $DBBASEDIR/db.$NSSLAPD_DB_LIB.br/userroot/entryrdn.db > $DBBASEDIR/entryrdn.$NSSLAPD_DB_LIB.ar
    dbscan -D ${NSSLAPD_DB_LIB:?} -f $DBBASEDIR/db.$NSSLAPD_DB_LIB.ar/userroot/entryrdn.db > $DBBASEDIR/entryrdn.$NSSLAPD_DB_LIB.br
}

/bin/rm -rf $DBBASEDIR
mkdir $DBBASEDIR
doit bdb
doit mdb
grep nsuniqueid $DBBASEDIR/entryrdn.*
```

**Comment by progier389** [comment]
Fixed by [PR 5498](https://github.com/389ds/389-ds-base/pull/5498)

**Comment by progier389** [comment]
Cherry-picked LMDB changes to 2.1 to fix the CLANG compile errors #5484:

6a78c9f5a0d9f922cdaf524712cdb6dd93a670d2 Issue 5408: lmdb import is slow (#5481)
   764a306e4..ce0ff7375  389-ds-base-2.2 -> 389-ds-base-2.2
   9476a78e3..539a29bdc  389-ds-base-2.1 -> 389-ds-base-2.1

4b82bf6d391ef12069e101ba2161ed0a25bbaba5 Issue 5490 - tombstone in entryrdn index with lmdb but not with bdb (#5498)
   ce0ff7375..067ec8134  389-ds-base-2.2 -> 389-ds-base-2.2
   539a29bdc..06bcbd913  389-ds-base-2.1 -> 389-ds-base-2.1

bb5df9d474d9cf9f224c90d8171b95e5042bedf5 Issue 5545 - A random crash in import over lmdb (#5546)
   067ec8134..acee4ca7c  389-ds-base-2.2 -> 389-ds-base-2.2
   06bcbd913..9d92f6a86  389-ds-base-2.1 -> 389-ds-base-2.1
